src/components/Navbar.astro

```diff
@@ -1,20 +1,86 @@
 ---
+import { Icon } from "astro-icon";
 export interface Props {
   theme?: "dark";
 }
 
-const { theme = '' } = Astro.props as Props;
+const { theme = "" } = Astro.props as Props;
 const navlinkClass = `navlink ${theme}`;
 ---
 
 <header>
-  <nav>
-    <a class={`${navlinkClass} skip-navlink`} href="#main">Skip navigation</a>
-    <a class={navlinkClass} href="/">Home</a>
-    <a class={navlinkClass} href="/#speakers">Speakers</a>
-    <a class={navlinkClass} href="/#agenda">Agenda</a>
+  <nav class="main-nav">
+    <div class="primary-links">
+      <a class={`${navlinkClass} skip-navlink`} href="#main">Skip navigation</a>
+      <a class={navlinkClass} href="/">Home</a>
+      <a class={navlinkClass} href="/#speakers">Speakers</a>
+      <a class={navlinkClass} href="/#agenda">Agenda</a>
+    </div>
+
+    <button class="hamburger" aria-label="Toggle menu">
+      <Icon name="mdi:menu" class={navlinkClass} />
+    </button>
   </nav>
 </header>
+
+<nav class="side-drawer">
+  <button class="close" aria-label="Close menu">
+    <Icon name="mdi:close" class="navlink" />
+  </button>
+  <section>
+    <div class="primary">
+      <a class="navlink" href="/">Home</a>
+      <a class="navlink" href="/#speakers">Speakers</a>
+      <a class="navlink" href="/#agenda">Agenda</a>
+    </div>
+
+    <div class="secondary">
+      <a class="navlink" href="/handbook">Handbook</a>
+      <a class="navlink" href="/code-of-conduct">Code of Conduct</a>
+      <a class="navlink" href="/privacy">Privacy Statement</a>
+      <a class="navlink" href="/terms-and-conditions">Terms and Conditions</a>
+      <a class="navlink" href="/accessibility ">Accessibility</a>
+      <a class="navlink" href="/media">Media Kit</a>
+      <a class="navlink" href="/meet-ecma ">Meet Ecma</a>
+    </div>
+  </section>
+</nav>
+<div id="backdrop"></div>
+
+<script>
+  const hamburgerBtn = document.querySelector(
+    ".hamburger"
+  ) as HTMLButtonElement;
+  const closeBtn = document.querySelector(".close") as HTMLButtonElement;
+  const backdrop = document.querySelector("#backdrop") as HTMLDivElement;
+  const sideDrawer = document.querySelector(".side-drawer") as HTMLElement;
+  const mainNav = document.querySelector(".main-nav") as HTMLElement;
+
+  const sideDrawerPrimaryLinks = document.querySelectorAll(
+    ".side-drawer .primary a"
+  ) as NodeListOf<HTMLAnchorElement>;
+
+  function onOpen() {
+    backdrop.classList.add("backdrop");
+    sideDrawer.classList.add("open");
+    mainNav.classList.add("hide");
```

Comment:
onOpen you could programatically focus the `close` button so you keep the keyboard navigation in swing